src/meta/src/hummock/compaction/level_selector.rs

```diff
@@ -221,22 +226,58 @@ impl DynamicLevelSelector {
         ctx
     }
 
-    fn set_compaction_config_for_level(&self, input: &mut SearchResult, base_level: usize) {
-        if input.select_level.level_idx == 0 && input.target_level.level_idx == 0 {
-            // TODO: reduce `target_file_size` after we implement sub-level.
-            input.target_file_size = self.config.min_compaction_bytes;
-        } else if input.select_level.level_idx == 0 {
-            input.target_file_size = self.config.target_file_size_base;
+    fn generate_task_by_compaction_config(
+        &self,
+        input: CompactionInput,
+        base_level: usize,
+    ) -> CompactionTask {
+        let target_file_size = if input.target_level == 0 {
+            self.config.target_file_size_base * 2
         } else {
-            assert!(input.target_level.level_idx as usize >= base_level);
-            input.target_file_size = self.config.target_file_size_base
-                << (input.target_level.level_idx as usize - base_level);
-        }
-        if input.target_level.level_idx == 0 {
-            input.compression_algorithm = self.config.compression_algorithm[0].clone();
+            assert!(input.target_level >= base_level);
+            self.config.target_file_size_base << (input.target_level - base_level)
+        };
+        let compression_algorithm = if input.target_level == 0 {
+            self.config.compression_algorithm[0].clone()
         } else {
-            let idx = input.target_level.level_idx as usize - base_level + 1;
-            input.compression_algorithm = self.config.compression_algorithm[idx].clone();
+            let idx = input.target_level - base_level + 1;
+            self.config.compression_algorithm[idx].clone()
+        };
+        let mut splits = vec![];
+        if input.input_levels.last().unwrap().table_infos.len() > 1 {
+            const SPLIT_RANGE_STEP: usize = 8;
```

Comment:
🤔 No. Because I do not think we need a high concurrency for sub-compaction. Most configuration would not be changed by user. And as I mentioned above, the most of time does not cost on sst build. So we can not support more concurrency to speed up compaction.

Reply:
Makes sense, so the minimum control granularity of compaction will be limited to compact_task ?